build-system/common/utils.js

```diff
@@ -111,7 +123,17 @@ function getFilesToCheck(globs, options = {}) {
     }
     return logFiles(filesChanged);
   }
-  return globby.sync(globs, options);
+  const {trackedOnly = false, ...globbyOptions} = options;
+  const files = globby.sync(globs, {
+    gitignore: trackedOnly,
+    ...globbyOptions,
+  });
+  if (trackedOnly) {
+    return getStdout(`git ls-files ${files.join(' ')}`)
+      .trim()
+      .split('\n');
+  }
```

Comment:
On looking at this more carefully, I think we should ignore untracked files and files in `.gitignore` under all circumstances, and make it the default behavior. (It doesn't make sense for any of the tasks that use this util function to look at untracked / gitignore'd files.)

Instead of post-processing the results from `globby.sync`, I think it's better to pass in an `ignore` option. Something like this might work:

```js
const untrackedFiles = getStdout(`git ls-files --others --exclude-standard`).trim().split('\n')
return globby.sync(globs, {ignore: untrackedFiles, gitignore: true, ...options});
```

Reply:
Great, this works!